packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -4312,7 +4312,9 @@ export class ContainerRuntime
 				pending,
 				pendingIdCompressorState,
 				pendingAttachmentBlobs,
-				sessionExpiryTimerStarted: this.garbageCollector.sessionExpiryTimerStarted,
+				sessionExpiryTimerStarted:
```

Comment:
I believe that now we should always return pending state even though we don't have pending blobs or messages, since updating the snapshot has collateral consequences like the one added here and we need to update that value even though we don't have pending state at the moment of stashing.

Reply:
Actually, that's not correct. If we don't have pending state, there is no point in using the serialized string to load from. I believe currently we go through all the process of process saved ops, replaying pending state and so on even though the string doesn't have any pending op (since we most surely have saved ops). There are tests which do that at least.

Maybe we need some kind of mechanism to inform serialize() user that we don't have pending state, therefore there is no point on saving the pending local state nor applying it later. For now we could just avoid all processes related to stashing if we load from such state.

Reply:
There is still a problem though, which is the rest of the container runtime pending object (pendingIdCompressorState and sessionExpiryTimerStarted). I wonder if we should update those values in a newly loaded container if there is no pending state. If that is true, IF in line 4305 should go away, bringing some potential issues. IF that is false, then our tests will need to provide pending ops or otherwise we won't see sessionExpiryTimerStarted